src/filesystem.cc

```diff
@@ -544,6 +544,8 @@ GCSFileSystem::GCSFileSystem(const GCSCredential& gs_cred)
       gs_cred.path_);
   if (creds) {
     client_ = gcs::Client(gcs::ClientOptions(*creds));
+  } else {
+    client_ = gcs::Client::CreateDefaultClient();
```

Comment:
```suggestion
    client_ = gcs::Client(
        gcs::ClientOptions(gcs::oauth2::CreateAnonymousCredentials()));
```

Reply:
It is better to use `gcs::Client(gcs::ClientOptions(gcs::oauth2::CreateAnonymousCredentials()));`, because the `gcs::Client::CreateDefaultClient();` is deprecated, and also the former one prints more informative message when the bucket is not properly setup for public read access.